zulip_bots/zulip_bots/bots/calendarbot/calendarbot.py

```diff
@@ -0,0 +1,134 @@
+from ics import Calendar, Event, Attendee
+from zulip_bots.lib import BotHandler
+from datetime import date,time,datetime,timedelta
+import logging
+from time import sleep
+from tempfile import TemporaryFile
+from dataclasses import dataclass
+import os
+
+logging.basicConfig(filename='calendarbot.log', encoding='utf-8', level=logging.DEBUG)
+
+#TODO: Take this from environment variables in the future
+bot_email = 'test-bot-bot@recurse.zulipchat.com'
+
+#create calendar_object
+
+@dataclass
+class MeetingDetails:
+    sender_id: int
+    invitees: list
+    meeting_start: datetime
+    meeting_end: datetime
+    name: str
+    summary: str
+    length_minutes: int
+
+
+
+class CalendarBotHandler(object):
+    """
+    This bot fucking loves meetings. 
+    """
+    def init(self):
+        self.message = {}
+        self.bot_handler
+
+    def usage(self):
+        return """Your description of the bot"""
+
+    def handle_message(self, message:dict, bot_handler):
+        self.message = message
+        self.bot_handler = bot_handler
+        message_content = message['content'].lower()
+        cal = Calendar()
```

Comment:
- Moved MeetingDetails as a nested class of CalendarBotHandler since its only relevant to that classes context.
- We're not really using the CalendarBot Handler as a dataclass (though thats loose concept) so I don't think we need to decorate it as one. I also don't know if that will affect the magic that Zulip uses to run the bot.
- I tried moving the calendar object to the class init method but am running into an issue where I can't instantiate it there. I suspect it might be something to do with zulip again "AttributeError: 'CalendarBotHandler' object has no attribute 'cal'". Moving it back to the handle_message method for now.